bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/json/JsonMarketplaceAddonService.java

```diff
@@ -0,0 +1,249 @@
+        this.eventPublisher = eventPublisher;
+    }
+
+    protected void unsetEventPublisher(EventPublisher eventPublisher) {
+        this.eventPublisher = null;
+    }
+
+    @Override
+    public String getId() {
+        return "jsonmarketplace";
+    }
+
+    @Override
+    public String getName() {
+        return "JSON 3rd Party Marketplace";
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void refreshSource() {
+        cachedAddons = (List<AddonEntryDTO>) marketplaceUrls.stream().map(urlString -> {
+            try {
+                URL url = new URL(urlString);
+                URLConnection connection = url.openConnection();
+                connection.addRequestProperty("Accept", "application/json");
+                try (Reader reader = new InputStreamReader(connection.getInputStream())) {
+                    Type type = TypeToken.getParameterized(List.class, AddonEntryDTO.class).getType();
+                    return (List<AddonEntryDTO>) Objects.requireNonNull(gson.fromJson(reader, type));
+                }
+            } catch (IOException e) {
+                return List.of();
+            }
+        }).flatMap(List::stream).filter(e -> showUnstable || "stable".equals(((AddonEntryDTO) e).maturity))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<Addon> getAddons(Locale locale) {
+        // TODO: fix that we need to refresh every time
+        refreshSource();
+        return cachedAddons.stream().map(this::fromAddonEntry).collect(Collectors.toList());
+    }
+
+    @Override
+    public Addon getAddon(String id, Locale locale) {
+        String remoteId = id.replace(ADDON_ID_PREFIX, "");
+        return cachedAddons.stream().filter(e -> remoteId.equals(e.id)).map(this::fromAddonEntry).findAny()
+                .orElse(null);
+    }
+
+    @Override
+    public List<AddonType> getTypes(Locale locale) {
+        return new ArrayList<>(TAG_ADDON_TYPE_MAP.values());
+    }
+
+    @Override
+    public void install(String id) {
+        Addon addon = getAddon(id, null);
+        for (MarketplaceAddonHandler handler : addonHandlers) {
+            if (handler.supports(addon.getType(), addon.getContentType())) {
+                if (!handler.isInstalled(addon.getId())) {
+                    try {
+                        handler.install(addon);
+                        postInstalledEvent(addon.getId());
+                    } catch (MarketplaceHandlerException e) {
+                        postFailureEvent(addon.getId(), e.getMessage());
+                    }
+                } else {
+                    postFailureEvent(addon.getId(), "Add-on is already installed.");
+                }
+                return;
+            }
+        }
+        postFailureEvent(id, "Add-on not known.");
+    }
+
+    @Override
+    public void uninstall(String id) {
+        Addon addon = getAddon(id, null);
```

Comment:
The `getAddon(String, Locale)` method may return `null`. We should add an additional check here to prevent NPEs.